### PR TITLE
pass arguments correctly in case of unicast

### PR DESCRIPTION
**Pull request by admd**

### CODE REVIEW

**Comment by chiaradiamarcelo** [comment]
Looks good! Thanks!

**Review by isbm — COMMENTED** [review]
I would suggest to delete all that code and use an existing framework for all this. There is also [almost a copypaste](https://github.com/akamensky/argparse) from famous Python's `argparse`, among tons of others...

File: server/unicast.go [hunk 1]
```diff
@@ -22,9 +22,13 @@ func (h *Unicast) DefaultMethod(r *http.Request, args *UnicastArgs, reply *struc
 		if err != nil {
 			log.Printf("Call error: %v", err)
 		}
-		argumentsForCall := make([]interface{}, len(args.ServerArgs)+1)
+		argumentsForCall := make([]interface{}, 0, len(args.ServerArgs)+1)
```
Comment (isbm) [inline server/unicast.go:25]: You don't need that at all. A zero-size will just be enough, just `append` it later:

```go
argumentsForCall := make([]interface{}, 0)
```

File: server/unicast.go [hunk 2]
```diff
@@ -22,9 +22,13 @@ func (h *Unicast) DefaultMethod(r *http.Request, args *UnicastArgs, reply *struc
...
 		url, sessionKey := apiSession.GetServerSessionInfoByServerID(args.HubSessionKey, args.ServerID)
-		argumentsForCall[0] = sessionKey
+		argumentsForCall = append(argumentsForCall, sessionKey)
```
Comment (isbm) [inline server/unicast.go:27]: ...and now  you use `append`. :wink:

File: server/unicast.go [hunk 3]
```diff
@@ -22,9 +22,13 @@ func (h *Unicast) DefaultMethod(r *http.Request, args *UnicastArgs, reply *struc
...
+
+		for _, srvArg := range args.ServerArgs {
+			argumentsForCall = append(argumentsForCall, srvArg)
+		}
```
Comment (isbm) [inline server/unicast.go:31]: Why not just `argumentsForCall = append(argumentsForCall, srvArg...)`, which is idiomatically correct instead of this?

Comment (admd) [reply server/unicast.go:31]: Thank you @isbm . I didn't know that. I will change it in my next PR.

**Comment by chiaradiamarcelo** [comment]
> I would suggest to delete all that code and use an existing framework for all this. There is also [almost a copypaste](https://github.com/akamensky/argparse) from famous Python's `argparse`, among tons of others...

We will take a look, thanks!

**Pull request merged**